experimental/collator/src/provider.rs

```diff
@@ -0,0 +1,266 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+// The reordering algorithms in this file are adapted from ICU4C and,
+// therefore, are subject to the ICU license as described in LICENSE.
+
+use crate::data_ce_to_primary;
+use crate::CollationElement;
+use crate::CollationElement32;
+use crate::Tag;
+use icu_char16trie::char16trie::Char16TrieIterator;
+use icu_codepointtrie::CodePointTrie;
+use icu_provider::{yoke, zerofrom};
+use zerovec::ule::AsULE;
+use zerovec::ule::RawBytesULE;
+use zerovec::ZeroVec;
+
+#[cfg(feature = "provider_serde")]
```

Comment:
call this feature just `serde`

Reply:
All the `provider_serde` stuff is copypaste from an earlier snapshot of `uprops` and can change to whatever the current pattern is.

Reply:
Yes please do, we've changed the `provider_serde` feature to `serde` and `datagen`, these are now final.

Reply:
Done.